systers_portal/meetup/views.py

```diff
@@ -19,3 +20,18 @@ class MeetupLocationList(ListView):
     template_name = "meetup/list_location.html"
     model = MeetupLocation
     paginate_by = 20
+
+
+class MeetupView(MeetupLocationMixin, DetailView):
```

Comment:
This view works, but some of the things are redundant. Basically we do everything manually in `get_queryset()` method, consequently `model` and `slug_url_kwarg` become unnecessary to set up. I believe if you remove them, it works the same.

Whether we remove them and keep overriding the queryset method or we can do something like that:

```python
class MeetupView(MeetupLocationMixin, DetailView):
  template_name = "meetup/meetup.html"
  model = MeetupLocation

  def get_context_data(self, **kwargs):
    context = super(MeetupView, self).get_context_data(**kwargs)
    context['meetup'] = Meetup.objects.filter(slug=self.kwargs['meetup_slug'], meetup_location=self.object)
    return context

  def get_meetup_location(self):
    return self.object
```

Here's how the first version can be simplified:

```python
class MeetupView(MeetupLocationMixin, DetailView):
  template_name = "meetup/meetup.html"

  def queryset(self):
    self.meetup_location = MeetupLocation.objects.get(slug=self.kwargs['slug'])
    return Meetup.objects.filter(slug=self.kwargs['meetup_slug'], meetup_location=self.meetup_location)

  def get_meetup_location(self):
    return self.meetup_location
```

**Not tested actually, but the idea should hold.**